code/ship/ship.cpp

```diff
@@ -4125,34 +4885,34 @@ int get_default_player_ship_index()
 // Goober5000 - this works better in its own function
 void ship_set_default_player_ship()
 {
-	int i;
-
 	// already have one
 	if(strlen(default_player_ship))
 		return;
 
+	auto it = Ship_info.cbegin();
```

Comment:
But there it gets set to cbegin() which overwrites the previous value so this only adds more variables to the scope which causes confusion.
IMO it would be better to declare two iterator variables in the respective for loops.